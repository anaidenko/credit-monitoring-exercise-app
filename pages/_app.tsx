import { AppProps } from 'next/app'
import { SWRConfig } from 'swr'

import useFallbackToMock from '@/libs/hooks/use-fallback-to-mock'
import useFetcher from '@/libs/hooks/use-fetcher'

import '@/styles/index.css'

export default function App({ Component, pageProps }: AppProps) {
  const [fetcher] = useFetcher()
  useFallbackToMock()

  return (
    <SWRConfig
      value={{
        fetcher,
        errorRetryCount: 1,
        revalidateOnFocus: false,
      }}
    >
      <Component {...pageProps} />
    </SWRConfig>
  )
}
