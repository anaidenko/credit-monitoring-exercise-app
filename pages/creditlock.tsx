import Head from 'next/head'

import Layout from '@/components/Layout'
import Header from '@/components/creditlock-page/Header'
import CreditFreezeCenter from '@/components/creditlock-page/CreditFreezeCenter'
import FactsSteps from '@/components/creditlock-page/FactsSteps'

export const CreditLock = (): JSX.Element => (
  <Layout>
    <Head>
      <title>Credit Lock Center</title>
    </Head>
    <main>
      <Header />
      <CreditFreezeCenter />
      <FactsSteps />
    </main>
  </Layout>
)

export default CreditLock
