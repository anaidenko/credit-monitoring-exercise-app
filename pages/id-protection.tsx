import Head from 'next/head'

import Layout from '@/components/Layout'
import Header from '@/components/id-protection-page/Header'
import NotificationInsurance from '@/components/id-protection-page/NotificationInsurance'
import NotificationLostWallet from '@/components/id-protection-page/NotificationLostWallet'
import NotificationRestorationServices from '@/components/id-protection-page/NotificationRestorationServices'

export const IdProtection = (): JSX.Element => (
  <Layout>
    <Head>
      <title>My Identity Manager</title>
    </Head>
    <main>
      <Header />
      <NotificationInsurance />
      <NotificationLostWallet />
      <NotificationRestorationServices />
    </main>
  </Layout>
)

export default IdProtection
