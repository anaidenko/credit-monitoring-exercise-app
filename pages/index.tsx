import Head from 'next/head'
import Link from 'next/link'

export const Home = (): JSX.Element => (
  <div className="container">
    <Head>
      <title>Exercise 1</title>
      <link rel="icon" href="/favicon.ico" />
    </Head>

    <main>
      <ul>
        <li>
          <Link href="/id-protection">
            <a>ID Protection</a>
          </Link>
        </li>
        <li>
          <Link href="/creditlock">
            <a>Credit Lock</a>
          </Link>
        </li>
      </ul>
    </main>
  </div>
)

export default Home
