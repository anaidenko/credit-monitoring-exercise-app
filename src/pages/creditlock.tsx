import useAuthToken from '@/components/hooks/useAuthToken'
import useFallbackToMock from '@/components/hooks/useFallbackToMock'
import BaseLayout from '@/components/layouts/BaseLayout'
import CreditFreezeCenter from '@/components/monitoring/pages/creditlock/CreditFreezeCenter'
import FactsSteps from '@/components/monitoring/pages/creditlock/FactsSteps'
import Header from '@/components/monitoring/pages/creditlock/Header'
import Loading from '@/components/shared/Loading'

type Props = {
  mock?: boolean
}

export const CreditLock = ({ mock }: Props): JSX.Element => {
  useFallbackToMock(mock)
  const [token, { error }] = useAuthToken()

  if (error) return <div>Error: {error.message}</div>
  if (!token) return <Loading />

  return (
    <BaseLayout>
      <Header />
      <CreditFreezeCenter />
      <FactsSteps />
    </BaseLayout>
  )
}

export default CreditLock
