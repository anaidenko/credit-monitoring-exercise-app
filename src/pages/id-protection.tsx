import { FunctionComponent } from 'react'

import useAuthToken from '@/components/hooks/useAuthToken'
import useFallbackToMock from '@/components/hooks/useFallbackToMock'
import BaseLayout from '@/components/layouts/BaseLayout'
import Header from '@/components/monitoring/pages/id-protection/Header'
import NotificationInsurance from '@/components/monitoring/pages/id-protection/NotificationInsurance'
import NotificationLostWallet from '@/components/monitoring/pages/id-protection/NotificationLostWallet'
import NotificationRestorationServices from '@/components/monitoring/pages/id-protection/NotificationRestorationServices'
import Loading from '@/components/shared/Loading'

type Props = {
  mock?: boolean
}

export const IdProtection: FunctionComponent<Props> = ({ mock }: Props): JSX.Element => {
  useFallbackToMock(mock)
  const [token, { error }] = useAuthToken()

  if (error) return <div>Error: {error.message}</div>
  if (!token) return <Loading />

  return (
    <BaseLayout>
      <Header />
      <NotificationInsurance />
      <NotificationLostWallet />
      <NotificationRestorationServices />
    </BaseLayout>
  )
}

export default IdProtection
