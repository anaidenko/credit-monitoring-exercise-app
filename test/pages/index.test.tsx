import React from 'react'

import { Home } from '../../pages/index'
import { render } from '../testUtils'

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Home />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  // xit('clicking button triggers alert', () => {
  //   const { getByText } = render(<Home />, {})
  //   window.alert = jest.fn()
  //   fireEvent.click(getByText('Test Button'))
  //   expect(window.alert).toHaveBeenCalledWith('With typescript and Jest')
  // })
})
