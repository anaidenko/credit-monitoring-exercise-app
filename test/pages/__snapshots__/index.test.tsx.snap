// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <main>
      <ul>
        <li>
          <a
            href="/id-protection"
          >
            ID Protection
          </a>
        </li>
        <li>
          <a
            href="/creditlock"
          >
            Credit Lock
          </a>
        </li>
      </ul>
    </main>
  </div>
</DocumentFragment>
`;
